src/tests/ftest/pool/evict.py

```diff
@@ -56,204 +53,160 @@ def connected_pool(self, hostlist, targets=None):
+        :avocado: tags=vm
+        :avocado: tags=pool
+        :avocado: tags=pool_evict,pool_evict_basic
         :avocado: tags=DAOS_5610
         """
         # Do not use self.pool. It will cause -1002 error when disconnecting.
-        pool = []
-        container = []
-        # non_pool_servers = []
-
-        # Target list is configured so that the pools are across all servers
-        # except the pool under test is created on half of the servers
-        pool_tgt = list(range(len(self.hostlist_servers)))
-        pool_tgt_ut = list(range(int(len(self.hostlist_servers)/2)))
-        tlist = [pool_tgt, pool_tgt, pool_tgt_ut]
-        pool_servers = [self.hostlist_servers[:len(tgt)] for tgt in tlist]
-        # non_pool_servers = [self.hostlist_servers[len(tgt):] for tgt in tlist]
-
-        # Create Connected TestPool
-        for count, target_list in enumerate(tlist):
-            pool.append(self.connected_pool(pool_servers[count], target_list))
-            # Commented out due to DAOS-3836.
-            # if len(non_pool_servers[count]) > 0:
-            #    self.assertFalse(
-            #        pool[count].check_files(non_pool_servers[count]),
-            #        "Pool # {} data detected on non pool servers {} ".format(
-            #            count+1, non_pool_servers[count]))
-
-            self.log.info(
-                "Pool # %s is connected with handle %s", count + 1,
-                pool[-1].pool.handle.value)
-
-            container.append(TestContainer(pool[-1]))
-            container[count].get_params(self)
-            container[count].create()
-            container[count].write_objects(target_list[-1])
-
-        try:
-            pool[-1].dmg.exit_status_exception = False
-            pool[-1].evict()
-        finally:
-            pool[-1].dmg.exit_status_exception = True
-
-        if pool[-1].dmg.result.exit_status != 0:
-            self.fail("Pool evict failed!")
-
-        for count in range(len(tlist)):
-            # Commented out due to DAOS-3836.
-            # # Check that all pool files still exist
-            # if pool[count].check_files(pool_servers[count]):
-            #    self.log.info(
-            #        "Pool # %s with UUID %s still exists",
-            #        count+1, pool[count].uuid)
-            # else:
-            #    self.fail(
-            #        "Pool # {} with UUID {} does not exists".format(
-            #            count+1, pool[count].uuid))
-
-            # Verify connection to pools with pool_query; pool that was evicted
-            # should fail the pool query because the handle was removed
+        pools = []
+        containers = []
+
+        # four_ranks stores the four ranks we have.
+        four_ranks = [0, 1, 2, 3]
+        # two_ranks_with is the two ranks that we'll create the third pool on.
+        two_ranks_with = [0, 1]
+        # two_ranks_without is the two ranks that we will not create a pool.
+        two_ranks_without = [2, 3]
```

Comment:
Actually, I think the previous way of using "all" and "half" is better than hardcoding "four" and "two". If we ever want to run this test on a smaller or larger cluster, the test would need to be updated. Is it easy to keep using "all" and "half"?

Reply:
Thanks for the comment. I noticed that it's easy to support arbitrary number of hosts, so updated the code.